packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {helpers, ShopperLogin, ShopperLoginTypes} from 'commerce-sdk-isomorphic'
+import jwtDecode from 'jwt-decode'
+import {ApiClientConfigParams} from '../hooks/types'
+import {BaseStorage, LocalStorage, CookieStorage} from './storage'
+
+type Helpers = typeof helpers
+interface AuthConfig extends ApiClientConfigParams {
+    redirectURI: string
+    proxy: string
+}
+
+interface JWTHeaders {
+    exp: number
+    iat: number
+}
+
+// this type is slightly different from ShopperLoginTypes.TokenResponse, reasons:
+// 1. TokenResponse is too generic (with & {[key:string]: any}), we need a more
+//    restrictive type to make sure type safe
+// 2. The refresh tokens are stored separately for guest and registered user. Instead
+//    of refresh_token, we have refresh_token_guest and refresh_token_registered
+type AuthDataKeys =
+    | 'access_token'
+    | 'customer_id'
+    | 'enc_user_id'
+    | 'expires_in'
+    | 'id_token'
+    | 'idp_access_token'
+    | 'refresh_token_guest'
+    | 'refresh_token_registered'
+    | 'token_type'
+    | 'usid'
+type AuthDataMap = Record<
+    AuthDataKeys,
+    {
+        storage: BaseStorage
+        key: string
+        callback?: () => void
+    }
+>
+
+/**
+ * This class is used to handle shopper authentication.
+ * It is responsible for initializing shopper session, manage access
+ * and refresh tokens on server/browser environments. As well as providing
+ * a mechanism to queue network calls before having a valid access token.
+ *
+ * @Internal
+ */
+class Auth {
+    private client: ShopperLogin<ApiClientConfigParams>
+    private redirectURI: string
+    private onClient = typeof window !== 'undefined'
+    private pending: Promise<ShopperLoginTypes.TokenResponse> | undefined
```

Comment:
```suggestion
    private pendingToken: Promise<ShopperLoginTypes.TokenResponse> | undefined
```
The name `pending` is unclear to me - my first thought would be it's a boolean. Something like `pendingToken` would make it more clear that it's eventual data, not an `isPending` flag.

Reply:
✅